### PR TITLE
(doc): Add Contributing.md

### LINKED ISSUE #344: add CONTRIBUTING.md
- single commit
- run `node lib/generate.js`

### PR DESCRIPTION
**Pull request by codydaig**
Closes #344

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+* Check the [Issues](https://github.com/mikedeboer/node-github/issues) on Github to ensure one doesn't already exist.
+* Clearly describe the issue, including the steps to reproduce the issue.
+* Please include links to the corresponding github documentation.
+
+## Making Changes
+
+* Create a topic branch from the master branch.
+* Check for unnecessary whitespace / changes with `git diff --check` before committing.
+* Keep git commit messages clear and appropriate
+	* If possible, please "squash" your commits to as few commits as possible/reasonable such as one commit for implementation, one for tests, and one for documentation before finally squashing to one commit when getting the LGTM from a collaborator.
```
Comment (kaizensoze) [inline CONTRIBUTING.md]: I'd just put squash to one commit from the start.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+* run `node lib/generate.js` and commit any files that appear. This will generate documentation and corresponding tests.
+
+
+## Submitting the Pull Request
+
+* Push your changes to your topic branch on your fork of the repo.
+* Submit a pull request from your topic branch to the master branch on the MEAN.JS repository.
```
Comment (kaizensoze) [inline CONTRIBUTING.md]: "on the node-github repository."

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+* Be sure to tag any issues your pull request is taking care of / contributing to.
+	* By adding "Closes #xyz" to a commit message will auto close the issue once the pull request is merged in.
```
Comment (kaizensoze) [inline CONTRIBUTING.md]: - Adding "Closes #xyz" to a commit message will auto close the issue once the pull request is merged in.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+* Small changes are usually accepted and merged in within a week (provided that 2 collaborators give the okay)
+* Larger changes usually spark further discussion and possible changes prior to being merged in.
```
Comment (kaizensoze) [inline CONTRIBUTING.md]: I'd just remove these last two bullet points.

**Comment by codydaig** [comment]
@kaizensoze Fixed above comments. I wrote the contributing doc for the meanjs repo so I used that as a template and missed a few lines. Should be good to go now.

**Pull request merged**